cpp/oneapi/dal/table/backend/homogen_table_impl.cpp

```diff
@@ -70,16 +70,8 @@ static void refer_origin_data(const array<DataSrc>& src,
     ONEDAL_ASSERT((src.get_count() - src_start_index) * sizeof(DataSrc) >=
                   dst_count * sizeof(DataDest));
 
-    if (src.has_mutable_data()) {
-        // TODO: in future, when table knows about mutability of its data this branch shall be
-        // available only for builders, not for tables.
-        auto start_pointer = reinterpret_cast<DataDest*>(src.get_mutable_data() + src_start_index);
-        dst.reset(src, start_pointer, dst_count);
-    }
-    else {
-        auto start_pointer = reinterpret_cast<const DataDest*>(src.get_data() + src_start_index);
-        dst.reset(src, start_pointer, dst_count);
-    }
+    auto start_pointer = reinterpret_cast<const DataDest*>(src.get_data() + src_start_index);
+    dst.reset(src, start_pointer, dst_count);
```

Comment:
Can you please check that the builder logic is not broken? I mean, pulling data from a builder does not require making a copy. At this moment, the builder shares one implementation with homogen table.

Reply:
Added test case for that
```c++
TEST("pull returns mutable data from homogen_table_builder") {
    constexpr std::int64_t row_count = 3;
    constexpr std::int64_t column_count = 2;
    float data[row_count * column_count] = { 1.0f, 2.0f, 3.0f, -1.0f, -2.0f, -3.0f };

    const auto ary = array<float>::wrap(data, row_count * column_count);
    const auto builder = detail::homogen_table_builder{}.reset(ary, row_count, column_count);
    const auto block = row_accessor<const float>{ builder }.pull();

    REQUIRE(block.has_mutable_data() == true);
}
```

Reply:
Thanks